src/angular/components/ciphers.component.ts

```diff
@@ -126,4 +128,31 @@ export class CiphersComponent {
         this.pagedCiphers = [];
         this.loadMore();
     }
+
+    selectAll(select: boolean) {
+        if (select) {
+            this.selectAll(false);
+        }
+        const selectCount = select && this.ciphers.length > this.maxCheckedCount
+            ? this.maxCheckedCount
+            : this.ciphers.length;
+        for (let i = 0; i < selectCount; i++) {
+            this.checkCipher(this.ciphers[i], select);
+        }
+    }
+
+    checkCipher(c: CipherView, select?: boolean) {
+        (c as any).checked = select == null ? !(c as any).checked : select;
+    }
+
+    getSelected(): CipherView[] {
+        if (this.ciphers == null) {
+            return [];
+        }
+        return this.ciphers.filter((c) => !!(c as any).checked);
+    }
+
+    getSelectedIds(): string[] {
+        return this.getSelected().map((c) => c.id);
```

Comment:
I dont think this needs to be in jslib since it still only applies to the web vault. the org ciphers componenet already inherits from the regular ciphers component where this lives in the web repo so it should be reusable already.

Reply:
Done